[#6] 통합테스트 및 단위테스트(음원 게시물)

cluejws opened the pull request:
## Description
- 음원 게시물 관련 통합테스트
- 음원 게시물 관련 서비스 단위테스트
- 음원 게시물 관련 레포지토리 단위테스트

cluejws committed:
[#6] refactor: module-core app 삭제

cluejws committed:
[#6] refactor: domain 생성자 추가

cluejws committed:
[#6] test: 음원 게시물 관련(컨트롤러)

- 음원 게시물 목록 조회

cluejws committed:
[#6] refactor: 음원 게시물 관련(컨트롤러) 테스트 수정

- 음원 게시물 목록 조회

cluejws committed:
[#6] feat: 조회 관련 예외 처리

cluejws committed:
[#6] test: 음원 게시물 관련(컨트롤러)

- 음원 게시물 조회

cluejws committed:
[#6] refactor: 요청 dto builder 생성자 적용

cluejws committed:
[#6] test: 음원 게시물 관련(컨트롤러)

- 음원 게시물 등록
- 음원 게시물 좋아요 달기

f-lab-lyan commented on module-api/src/test/java/com/soundie/post/controller/PostControllerTest.java:
```diff
@@ -0,0 +1,159 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.soundie.member.domain.Member;
+import com.soundie.post.domain.Post;
+import com.soundie.post.dto.*;
```
이거 풀어서 모든 import가 다 보일 수 있도록 설정을 변경해 주세요. :-)

cluejws committed:
[#6] refactor: 음원 게시물 관련(컨트롤러) 테스트 any() 적용

cluejws committed:
[#6] test: 음원 게시물 관련(컨트롤러) 테스트 support 적용

cluejws committed:
[#6] test: 음원 게시물 관련(서비스) 테스트

- 음원 게시물 목록 조회

cluejws committed:
[#6] test: 음원 게시물 관련(서비스) 테스트

- 음원 게시물 목록 조회
- 음원 게시물 조회

cluejws committed:
[#6] test: 음원 게시물 관련(서비스) 테스트

- 음원 게시물 조회 실패

cluejws committed:
[#6] test: 음원 게시물 관련(서비스) 테스트

- 음원 게시물 등록 성공

f-lab-lyan commented on module-api/src/test/java/com/soundie/post/controller/PostControllerTest.java:
```diff
@@ -0,0 +1,155 @@
+
+        // when // then
+        mockMvc.perform(
+                        get("/api/posts/" + postId)
```
```suggestion
                        get("/api/posts/" + postId + "?memberId=1")
```

cluejws committed:
[#6] remove: 불필요한 앱 삭제

cluejws committed:
[#6] refactor: 요청 dto 생성자 적용

- builder 생성자 -> 생성자

cluejws committed:
[#6] refactor: 음원 게시물 관련(서비스) 테스트 Mocking 설정

cluejws committed:
[#6] test: 음원 게시물 관련 통합 테스트

cluejws committed:
[#6] refactor: 음원 게시물 관련(서비스) 테스트 Mocking

cluejws committed:
[#6] test: 음원 게시물 관련(서비스) 단위 테스트

- 음원 게시물 등록 실패
   - 유효하지 않은 memberId
- 음원 게시물 좋아요 달기 성공
- 음원 게시물 좋아요 취소 성공
- 음원 게시물 좋아요 달기 실패
   - 유효하지 않은 memberId
   - 유효하지 않은 postId

cluejws committed:
[#6] refactor: 음원 게시물 관련(서비스) 단위 테스트 명칭 수정

cluejws committed:
[#6] refactor: 회원 관련(레포지토리) 추상화

cluejws committed:
[#6] refactor: 댓글 관련(레포지토리) 추상화

cluejws committed:
[#6] refactor: 채팅방 관련(레포지토리) 추상화

cluejws committed:
[#6] refactor: 알림 관련(레포지토리) 추상화

cluejws committed:
[#6] refactor: 음원 게시물 관련(레포지토리) 추상화

cluejws committed:
[#6] test: 음원 게시물 관련(레포지토리) 단위 테스트

cluejws committed:
[#6] refactor: 통합 테스트를 위한 TestSupport 파일 이동

cluejws committed:
[#6] style: 코드 포맷팅

cluejws committed:
[#6] test: 댓글 관련 통합 테스트

cluejws committed:
[#6] fix: 음원 게시물 및 댓글 관련(서비스) 예외 처리

cluejws committed:
[#6] refactor: fixture 파일 패키지 이동

cluejws committed:
[#6] test: 댓글 관련(서비스) 단위 테스트

cluejws committed:
[#6] refactor: fixture static 표현

cluejws committed:
[#6] test: 댓글 관련(레포지토리) 단위 테스트

cluejws committed:
[#6] refactor: 음원 게시물 관련(레포지토리) 단위 테스트 명칭 수정

cluejws committed:
[#6] fix: 음원 게시물 관련(레포지토리) 단위 테스트 실패 수정

- null -> 0L

cluejws committed:
[#6] refactor: 댓글 관련(서비스) 단위 테스트 수정

- 분기문 제거

f-lab-lyan reviewed (approved):
괜찮은 것 같습니다! 고생하셨어요.

jongyoul commented on module-api/src/test/java/com/soundie/comment/controller/CommentControllerTest.java:
```diff
@@ -0,0 +1,70 @@
+                "댓글 내용"
+        );
+    }
+}
```
나중에 이 파일 고칠 일이 있으시면, 마지막에 새줄 추가 부탁 드릴께요.

jongyoul commented on module-core/src/main/java/com/soundie/comment/service/CommentService.java:
```diff
@@ -20,34 +22,39 @@
-    private final CommentRepository commentRepository;
-    private final MemberRepository memberRepository;
-    private final PostRepository postRepository;
+    private final MemoryCommentRepository commentRepository;
```
이건 인터페이스로 안 받은 이유가 있는것인가요?

jongyoul commented on module-core/src/main/java/com/soundie/post/repository/MemoryPostLikeRepository.java:
```diff
@@ -0,0 +1,61 @@
+import com.soundie.post.domain.PostLike;
+import org.springframework.stereotype.Repository;
+
+import java.util.*;
```
설정을 바꿔서 모두 보이도록 해주세요.
